lib/Doctrine/ORM/Query/Lexer.php

```diff
@@ -164,7 +164,7 @@ protected function getType(&$value)
                 return self::T_STRING;
 
             // Recognize identifiers
-            case (ctype_alpha($value[0]) || $value[0] === '_'):
+            case (ctype_alpha($value[0]) || $value[0] === '_' || $value[0] === '\\'):
```

Comment:
Ouch. I was not aware of that and in fact it *was* possible to use a FQCN with a leading `\` with no problems – probably due to the parser inaccuracy addressed here. 

Just change one of the tests in `SelectSqlGenerationTest` on `master` and see yourself :-).